core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/function/Function.java

```diff
@@ -9,14 +9,45 @@
 
 import org.eclipse.rdf4j.model.Value;
 import org.eclipse.rdf4j.model.ValueFactory;
+import org.eclipse.rdf4j.query.algebra.evaluation.TripleSource;
 import org.eclipse.rdf4j.query.algebra.evaluation.ValueExprEvaluationException;
 
 /**
+ * A query function, which can be a built-in function in the query language, or a custom function as documented in the
+ * <a href="https://www.w3.org/TR/sparql11-query/">SPARQL 1.1 Query Language Recommendation</a>.
+ *
  * @author Arjohn Kampman
+ * @author Jeen Broekstra
+ * @see FunctionRegistry
  */
 public interface Function {
 
-	public String getURI();
+	String getURI();
 
-	public Value evaluate(ValueFactory valueFactory, Value... args) throws ValueExprEvaluationException;
+	/**
+	 * Evaluate the function over the supplied input arguments, using the supplied {@link ValueFactory} to produce the
+	 * result.
+	 *
+	 * @param valueFactory a {@link ValueFactory} to use for producing the function result.
+	 * @param args         the function input arguments.
+	 * @return the function result value.
+	 * @throws ValueExprEvaluationException
+	 * @deprecated since 3.3.0. Use {@link #evaluate(TripleSource, Value...)} instead.
```

Comment:
Could you add to the depreciation warning that the value factory can be retrieved from the triple source?